protokube/pkg/protokube/volume_mounter.go

```diff
@@ -105,54 +105,51 @@ func (k *VolumeMountController) safeFormatAndMount(device string, mountpoint str
 	}
 	glog.Infof("Found device %q", device)
 
-	// If we are containerized, we still first SafeFormatAndMount in our namespace
-	// This is because SafeFormatAndMount doesn't seem to work in a container
-	safeFormatAndMount := &mount.SafeFormatAndMount{Interface: mount.New(""), Exec: mount.NewOsExec()}
+	safeFormatAndMount := &mount.SafeFormatAndMount{}
+
+	if Containerized {
+		// Build mount & exec implementations that execute in the host namespaces
+		safeFormatAndMount.Interface = mount.NewNsenterMounter()
+		safeFormatAndMount.Exec = NewNsEnterExec()
+
+		// Note that we don't use pathFor for operations going through safeFormatAndMount,
+		// because NewNsenterMounter and NewNsEnterExec will operate in the host
+	} else {
+		safeFormatAndMount.Interface = mount.New("")
+		safeFormatAndMount.Exec = mount.NewOsExec()
```

Comment:
Is this a dry refactor for the non containerized os? If not what have we tested it on?